### PR TITLE
F UI diet plan presets

### PR DESCRIPTION
**Pull request by Noam-Moskowitz**

### CODE REVIEW

File: frontend/src/pages/ViewDietPlanPresetPage.tsx [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+        .catch((err) => {
+          toast.error("אופס, נתקלנו בבעיה!", { description: err.response.data.message });
+          console.error("error", err);
+        });
+
+ }
+
+ const editDietPlanPreset = (dietPlan:IDietPlan)=>{
+  if (!dietPlan || !id || !presetName) return;
+  if (!id) return;
```
Comment (mike16345) [inline frontend/src/pages/ViewDietPlanPresetPage.tsx:51]: You are already checking this

File: frontend/src/pages/ViewDietPlanPresetPage.tsx [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+        .catch((err) => {
+          toast.error("אופס, נתקלנו בבעיה!", { description: err.response.data.message });
+          console.error("error", err);
+        });
+ }
+
+
+  useEffect(() => {
+    setIsLoading(true)
+    if (id) {
```
Comment (mike16345) [inline frontend/src/pages/ViewDietPlanPresetPage.tsx:71]: Put a space before the start of if statements/loops/functions etc...

File: frontend/src/pages/ViewDietPlanPresetPage.tsx [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+export const ViewDietPlanPresetPage = () => {
+  const { id } = useParams();
+
+  const { getDietPlanPreset, updateDietPlanPreset, addDietPlanPreset } = useDietPlanPresetApi();
+
+   const [isNewPlan,setIsNewPlan] = useState<boolean>(false)
+
+    const [dietPlan, setDietPlan] = useState<IDietPlan>();
+
+    const [presetName,setPresetName]=useState<string>()
```
Comment (mike16345) [inline frontend/src/pages/ViewDietPlanPresetPage.tsx:23]: Do you need a state for this?

File: frontend/src/pages/ViewDietPlanPresetPage.tsx [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+import {  removeIdsAndVersions } from "@/utils/dietPlanUtils";
+
+export const ViewDietPlanPresetPage = () => {
+  const { id } = useParams();
+
+  const { getDietPlanPreset, updateDietPlanPreset, addDietPlanPreset } = useDietPlanPresetApi();
+
+   const [isNewPlan,setIsNewPlan] = useState<boolean>(false)
+
+    const [dietPlan, setDietPlan] = useState<IDietPlan>();
```
Comment (mike16345) [inline frontend/src/pages/ViewDietPlanPresetPage.tsx:21]: Typing is incorrect. Typescript infers types by the initial value. Your initial value is undefined. So typing it as IDietPlan is incorrect. Either give it a default diet plan or explicitly define it as null.

Comment (mike16345) [reply frontend/src/pages/ViewDietPlanPresetPage.tsx:21]: Same goes for preset name,

File: frontend/src/pages/ViewDietPlanPresetPage.tsx [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+
+    const [presetName,setPresetName]=useState<string>()
+
+    const [isLoading,setIsLoading]=useState<boolean>(false)
```
Comment (mike16345) [inline frontend/src/pages/ViewDietPlanPresetPage.tsx:25]: There is no need to type this as boolean. Typescript infers that the type is boolean based off the value you set. So it knows that it is a boolean already

File: frontend/src/pages/ViewDietPlanPresetPage.tsx [hunk 6]
```diff
@@ -0,0 +1,120 @@
...
+
+    const [error,setError]=useState<string>()
```
Comment (mike16345) [inline frontend/src/pages/ViewDietPlanPresetPage.tsx:27]: Read above comments. Also apply all these changes anywhere else in the project you may have done the same thing

File: frontend/src/routes/AppRoutes.tsx [hunk 7]
```diff
@@ -19,7 +20,9 @@ export const AppRoutes = () => {
         <Route path="/workoutPlans" element={<WorkoutsTemplatePage />} />
         <Route path="/presets/workoutPlans/" element={<WorkoutPreset />} />
         <Route path="/presets/workoutPlans/:id" element={<WorkoutPreset />} />
-        <Route path="/dietPlans" element={<DietPlanTemplatePage />} />
+        <Route path="/presets/dietPlans/" element={<ViewDietPlanPresetPage/>} />
```
Comment (mike16345) [inline frontend/src/routes/AppRoutes.tsx:23]: Open file called PresetsRoutes and move them all there.

File: frontend/src/components/ui/skeletons/TemplateTabsSkeleton.tsx
Comment (mike16345) [inline frontend/src/components/ui/skeletons/TemplateTabsSkeleton.tsx:1]: I like the look a lot

Comment (mike16345) [reply frontend/src/components/ui/skeletons/TemplateTabsSkeleton.tsx:1]: Hard  to see in light mode though

File: frontend/src/hooks/useWorkoutPlanPresetsApi.ts [hunk 8]
```diff
@@ -1,42 +1,29 @@
 import { deleteItem, fetchData, sendData, updateItem } from "@/API/api";
-import { ICompleteWorkoutPlan, IWorkoutPlanPreset } from "@/interfaces/IWorkoutPlan";
+import { IWorkoutPlanPreset } from "@/interfaces/IWorkoutPlan";
 
-const WORKOUT_PLAN_ENDPOINT = "workoutPlansPresets/";
+const WORKOUT_PLAN_PRESETS_ENDPOINT = "presets/workoutPlans/";
 
-export const useWorkoutPlanPresetApi = () => {/* 
-  
-
```
Comment (mike16345) [inline frontend/src/hooks/useWorkoutPlanPresetsApi.ts:8]: All this was stuff i deleted but notice how much space and comments there were. Before you open PRs go through the files you've worked on and make sure they are formatted properly and that you've deleted any unnecessary comments and console.logs etc....

**Review by mike16345 — CHANGES_REQUESTED** [review]

File: frontend/src/components/ui/Loader.tsx [hunk 9]
```diff
@@ -0,0 +1,22 @@
+import React from 'react'
+
+
```
[the remark this answers is not in the document]
Comment (Noam-Moskowitz) [reply frontend/src/components/ui/Loader.tsx:3]: i ammmmmm

File: frontend/src/components/templates/TemplateTabs.tsx [hunk 10]
```diff
@@ -50,6 +50,14 @@ const TemplateTabs: React.FC<TemplateTabsProps> = ({ tabs }) => {
         navigate(`/presets/workoutPlans/`)
       }
     }
+
+    if (selectedForm ===`dietPlans`) {
```
[the remark this answers is not in the document]
Comment (mike16345) [reply frontend/src/components/templates/TemplateTabs.tsx:50]: Do you  need it though

File: frontend/src/components/ui/Loader.tsx
Comment (mike16345) [inline frontend/src/components/ui/Loader.tsx:1]: This file ain't formatted

File: frontend/src/enums/ErrorMessages.ts [hunk 11]
```diff
@@ -0,0 +1 @@
+export const ERROR_MESSAGE_TITLE = `אופס, נתקלנו בבעיה!`;
```
Comment (mike16345) [inline frontend/src/enums/ErrorMessages.ts:1]: This is a const not an Enum

File: frontend/src/interfaces/IDietPlan.ts [hunk 12]
```diff
@@ -20,6 +20,11 @@ export interface IDietPlan {
   meals: IMeal[];
   totalCalories?: number;
 }
```
Comment (mike16345) [inline frontend/src/interfaces/IDietPlan.ts:22]: add space here

File: frontend/src/pages/DietPlanTemplatePage.tsx [hunk 13]
```diff
@@ -1,98 +1,118 @@
...
-
-    const [proteinMenuState, setProteinMenuState] = useState<IMenuItem[]>();
-    const [carbsMenuState, setCarbsMenuState] = useState<IMenuItem[]>();
-    const [VegetableMenuState, setVegetableMenuState] = useState<IMenuItem[]>();
-    const [fatsMenueState, setFatsMenueState] = useState<IMenuItem[]>();
+  const [dietPlanPresetsState, setDietPlanPresetsState] = useState<IDietPlan[]>();
+  const [proteinMenuState, setProteinMenuState] = useState<IMenuItem[]>();
+  const [carbsMenuState, setCarbsMenuState] = useState<IMenuItem[]>();
+  const [VegetableMenuState, setVegetableMenuState] = useState<IMenuItem[]>();
+  const [fatsMenueState, setFatsMenueState] = useState<IMenuItem[]>();
```
Comment (mike16345) [inline frontend/src/pages/DietPlanTemplatePage.tsx:17]: There is no need to write state in the variable name.

File: frontend/src/pages/DietPlanTemplatePage.tsx [hunk 14]
```diff
@@ -1,98 +1,118 @@
...
-                value: `fatsItems`,
-                btnPrompt: `הוסף שומנים`,
-                state: fatsMenueState || [],
-                sheetForm: `fats`,
-                deleteFunc: deleteMenuItem
-            },
-        ]
-    }
+  const [isLoading, setIsLoading] = useState(false);
+  const [error, setError] = useState<string>();
```
Comment (mike16345) [inline frontend/src/pages/DietPlanTemplatePage.tsx:20]: explicilty set this to null

File: frontend/src/pages/DietPlanTemplatePage.tsx [hunk 15]
```diff
@@ -1,98 +1,118 @@
...
+        deleteFunc: deleteMenuItem,
+      },
+    ],
+  };
 
+  useEffect(() => {
+    setIsLoading(true);
+    getAllDietPlanPresets()
+      .then((res) => setDietPlanPresetsState(res))
+      .catch((err) => setError(err));
```
Comment (mike16345) [inline frontend/src/pages/DietPlanTemplatePage.tsx:88]: place spaces between function calls

File: frontend/src/pages/ViewDietPlanPage.tsx [hunk 16]
```diff
@@ -1,138 +1,129 @@
...
-  const [isNewPlan, setIsNewPlan] = useState(true);
-  const [mealToDelete, setMealToDelete] = useState<number | null>(null);
+  const { getAllDietPlanPresets } = useDietPlanPresetApi();
 
-  const [dietPlan, setDietPlan] = useState<IDietPlan | null>(null);
+  const [isNewPlan, setIsNewPlan] = useState(false);
 
-  const handleAddMeal = () => {
-    if (!dietPlan) return;
+  const [dietPlan, setDietPlan] = useState<IDietPlan>();
```
Comment (mike16345) [inline frontend/src/pages/ViewDietPlanPage.tsx:29]: default plan or null

File: frontend/src/pages/ViewDietPlanPage.tsx [hunk 17]
```diff
@@ -1,138 +1,129 @@
...
 
-    setDietPlan({ ...dietPlan, meals: [...dietPlan.meals, defaultMeal] });
-  };
+  const [presetList, setPresetList] = useState<IDietPlanPreset[]>();
```
Comment (mike16345) [inline frontend/src/pages/ViewDietPlanPage.tsx:31]: empty array or null

File: frontend/src/pages/ViewDietPlanPresetPage.tsx [hunk 18]
```diff
@@ -0,0 +1,116 @@
...
+      })
+      .catch((err) => {
+        toast.error(ERROR_MESSAGE_TITLE, { description: err.response.data.message });
+        console.error("error", err);
+      });
+  };
+
+  const editDietPlanPreset = (dietPlan: IDietPlan) => {
+    if (!dietPlan || !id || !presetName) return;
+    if (!id) return;
```
Comment (mike16345) [inline frontend/src/pages/ViewDietPlanPresetPage.tsx:50]: didnt i leave a comment here?

File: frontend/src/pages/ViewDietPlanPresetPage.tsx [hunk 19]
```diff
@@ -0,0 +1,120 @@
...
+import { Label } from "@/components/ui/label";
+import {  removeIdsAndVersions } from "@/utils/dietPlanUtils";
+
+export const ViewDietPlanPresetPage = () => {
+  const { id } = useParams();
+
+  const { getDietPlanPreset, updateDietPlanPreset, addDietPlanPreset } = useDietPlanPresetApi();
+
+   const [isNewPlan,setIsNewPlan] = useState<boolean>(false)
+
```
[the remark this answers is not in the document]
Comment (mike16345) [reply frontend/src/pages/ViewDietPlanPresetPage.tsx:21]: ??

**Review by mike16345 — CHANGES_REQUESTED** [review]

File: frontend/src/pages/DietPlanTemplatePage.tsx [hunk 20]
```diff
@@ -1,98 +1,118 @@
...
+    getMenuItems(`carbs`)
+      .then((res) => setCarbsMenuState(res))
+      .catch((err) => setError(err));
+    getMenuItems(`vegetables`)
+      .then((res) => setVegetableMenuState(res))
+      .catch((err) => setError(err));
+    getMenuItems(`fats`)
+      .then((res) => setFatsMenueState(res))
+      .catch((err) => setError(err));
```
Comment (mike16345) [inline frontend/src/pages/DietPlanTemplatePage.tsx:100]: Don't you have a function that gets you all the items at once?

File: frontend/src/pages/DietPlanTemplatePage.tsx [hunk 21]
```diff
@@ -1,98 +1,118 @@
...
 
+    setTimeout(() => {
+      setIsLoading(false);
```
Comment (mike16345) [inline frontend/src/pages/DietPlanTemplatePage.tsx:103]: you can lower this to a second

File: frontend/src/pages/ErrorPage.tsx
Comment (mike16345) [inline frontend/src/pages/ErrorPage.tsx:1]: this page aint formatted

File: frontend/src/hooks/useWorkoutPlanPresetsApi.ts (hunk 8 above)
Comment (mike16345) [reply frontend/src/hooks/useWorkoutPlanPresetsApi.ts:8]: @Noam-Moskowitz Did you see this? If yes mark it as resolved or let me know and i will.

File: frontend/src/pages/DietPlanTemplatePage.tsx (hunk 20 above)
Comment (Noam-Moskowitz) [reply frontend/src/pages/DietPlanTemplatePage.tsx]: Yes, but then I would have to sift them before dispersing them into the proper states so this felt easier and better to me

File: frontend/src/enums/ErrorMessages.ts [hunk 22]
```diff
@@ -0,0 +1,3 @@
+export const ERROR_MESSAGES = {
```
Comment (mike16345) [inline frontend/src/enums/ErrorMessages.ts:1]: Search up how to write  an enum. THis is technically fine but it is still not an Enum

**Review by mike16345 — APPROVED** [review]

**Pull request merged**